src/router.ts

```diff
@@ -20,12 +22,15 @@ const REQUIRE_MAIN_FILE = path.dirname(require.main?.filename || process.cwd())
  * @param app An express app or router instance
  * @param options An options object (optional)
  */
-const createRouter = <T = Express>(app: T, options: Options = {}): T => {
+const createRouter = async <T extends ExpressLike = ExpressLike>(
+  app: T,
+  options: Options = {}
+): Promise<T> => {
   const files = walkTree(
     options.directory || path.join(REQUIRE_MAIN_FILE, "routes")
   )
 
-  const routes = generateRoutes(files)
+  const routes = await generateRoutes(files)
```

Comment:
We _could_ make this `.then` able instead of awaited, but I'm not sure what side effects or if race conditions get introduced